nixos/modules/services/web-apps/gancio.nix

```diff
@@ -0,0 +1,280 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.gancio;
+  settingsFormat = pkgs.formats.json { };
+  inherit (lib)
+    mkEnableOption
+    mkPackageOption
+    mkOption
+    types
+    literalExpression
+    mkIf
+    optional
+    mapAttrsToList
+    concatStringsSep
+    concatMapStringsSep
+    getExe
+    mkMerge
+    mkDefault
+    ;
+in
+{
+  options.services.gancio = {
+    enable = mkEnableOption "Gancio, a shared agenda for local communities";
+
+    package = mkPackageOption pkgs "gancio" { };
+
+    plugins = mkOption {
+      type = with types; listOf package;
+      default = [ ];
+      example = literalExpression "[ pkgs.gancioPlugins.telegram-bridge ]";
+      description = ''
+        Paths of gancio plugins to activate (linked under $WorkingDirectory/plugins/).
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      description = "The user (and PostgreSQL database name) used to run the gancio server";
+      default = "gancio";
+    };
+
+    settings = mkOption rec {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+        options = {
+          hostname = mkOption {
+            type = types.str;
+            description = "The domain name under which the server is reachable.";
+          };
+          baseurl = mkOption {
+            type = types.str;
+            default = "";
+            example = "/gancio";
+            description = "The URL path under which the server is reachable.";
+          };
+          server = {
+            host = mkOption {
+              type = types.str;
+              default = "localhost";
+              example = "::";
+              description = ''
+                The address (IPv4, IPv6 or DNS) for the gancio server to listen on.
+              '';
+            };
+            port = mkOption {
+              type = types.port;
+              default = 13120;
```

Comment:
Ports shouldn't be hardcoded, there are way too little possible values to avoid conflicts.

There seems to be the option to run nuxt with a unix domain socket, that might be the better option. Or we force the user to set this.

Reply:
This is the default port for gancio, this is fine.
https://framagit.org/les/gancio/-/blob/master/server/config.js#L11

Reply:
It might be on their docs section, but it is not the reserved port: https://en.wikipedia.org/wiki/List_of_TCP_and_UDP_port_numbers

Debugging conflicting port numbers is hard, please let's not do this (and just use unix domain sockets, I'm working on a solution).

Reply:
> Debugging conflicting port numbers is hard

Then set it explicitly locally. And normally you should know what port it's running if you're going to add some service to your config.

Reply:
Point me to the docs then, that clarify on this. Just telling me that what I need to know is not a technical reason. And stop marking threads as resolved that clearly aren't.

From a quick grep, it looks like there are a lot more services in `nixos/modules/services/web-apps/` that set no default for port than who do.

Reply:
> Just telling me that what I need to know is not a technical reason

what does that sentence even mean?

> Point me to the docs then, that clarify on this.

You've pulled your own guidelines out of thin air and are nitting a PR that's half a year old over it. That's actively hostile to contributors and I've no intention of encouraging such behaviour in nixpkgs. If you have a followup PR please feel free to create one rather than blocking this one over something that just about every module already does.

> And stop marking threads as resolved that clearly aren't.

Nits should not be entertained.

Reply:
This interaction is clearly not leading anywhere.

My goal is to get this PR merged. I don't care enough about this detail and leave it to the judgement of the author.